lesson16/main.js

```diff
@@ -0,0 +1,111 @@
+// json = https://mocki.io/v1/d3176c84-479a-47a3-a8b6-7d597b55d8b4
+const ul = document.getElementById("js-ul");
+const li = document.createElement("li");
+const contetns = document.getElementsByClassName("tab-content");
+const tabs = document.getElementsByClassName("tab-item");
+
+const displayInfo = (info) => {
+  const li = document.createElement("li");
+  if (typeof info === "string") {
+    li.textContent = info;
+    ul.appendChild(li);
+    return console.log(info);
+  }
+  throw new Error("Please pass a string value");
+};
+
+const displayErrorInfo = (error) => {
+  console.log(error);
+  li.textContent = `Failed to get data from server`;
+  ul.appendChild(li);
+};
+
+const getData = async (api) => {
+  let response = null;
+  try {
+    response = await fetch(api);
+  } catch (error) {
+    displayErrorInfo(error);
+  }
+  if (response.ok) {
+    return response.json();
+  } else {
+    displayErrorInfo(response);
+  }
+};
+
+const checkResponse = async (callBackFunction) => {
+  let response = await getData(
+    "https://mocki.io/v1/40cbd0f7-2a64-4b95-a991-57ff982901a8"
+  );
+  if (response.data) {
+    callBackFunction(response.data);
+  } else {
+    displayInfo("no data available");
+  }
+};
+
+const createTabs = (items) => {
+  ul.classList = "tab-menu";
+  const fragment = new DocumentFragment();
+  for (let item of items) {
+    const li = document.createElement("li");
+    li.textContent = item.category;
+    li.classList = "tab-item";
+    fragment.appendChild(li);
+  }
+  ul.appendChild(fragment);
+  activateFirstTab();
+};
+
+const activateFirstTab = () => {
+  tabs[0].classList.add("active");
+};
```

Comment:
IMO
I felt that this place did not need to be a function.It's just one line.

```suggestion
  ul.appendChild(fragment);
  tabs[0].classList.add("active");
};

//const activateFirstTab = () => {
//  tabs[0].classList.add("active");
//};
```

Reply:
Thank you for the comment Chihiro-san,
I thought it would be better to have a function as something similar process was used in another function. But the code you suggested made it look more readable 
Please see the update
59fb045fc215cf1bab0148169691a59cf80422a1